File: tests/fixtures/_highlight-corpus/sample.jl
# Julia sample
module Sample

mutable struct Counter
    value::Int
    Counter() = new(0)
end

function increment!(c::Counter, n::Integer = 1)
    c.value += n
    return c.value
end

function count_bytes(path::AbstractString)::Int
    counter = Counter()
    open(path, "r") do f
        for line in eachline(f)
            if !isempty(line)
                increment!(counter, length(line))
            end
        end
    end
    return counter.value
end

function main(argv::Vector{String} = ARGS)
    path = isempty(argv) ? "input.txt" : argv[1]
    total = count_bytes(path)
    println("total bytes: $total")
    return 0
end

end  # module Sample

if abspath(PROGRAM_FILE) == @__FILE__
    Sample.main()
end

File: tests/fixtures/_highlight-corpus/sample.ex
# Elixir sample
defmodule Counter do
  @moduledoc "A small counter."

  defstruct value: 0

  @spec increment(t :: %__MODULE__{}, integer()) :: %__MODULE__{}
  def increment(%__MODULE__{value: v}, n \\ 1), do: %__MODULE__{value: v + n}

  @spec value(t :: %__MODULE__{}) :: integer()
  def value(%__MODULE__{value: v}), do: v
end

defmodule Sample do
  def count_bytes(path) do
    path
    |> File.stream!()
    |> Enum.reduce(%Counter{}, fn line, acc ->
      trimmed = String.trim_trailing(line, "\n")
      case trimmed do
        "" -> acc
        s  -> Counter.increment(acc, String.length(s))
      end
    end)
    |> Counter.value()
  end

  def main(args \\ []) do
    path = List.first(args, "input.txt")
    total = count_bytes(path)
    IO.puts("total bytes: #{total}")
  end
end

Sample.main(System.argv())
